### PR TITLE
ci: migrate yamllint to GitHub Actions

### PR DESCRIPTION
**Pull request by tnevrlka**
[STONEBLD-2791](https://issues.redhat.com/browse/STONEBLD-2791)
- Container image used in yaml-lint task is outdated and using it causes rate-limiting issues with docker.io
- Remove the yaml-lint task to task-lint and remove yamllint part
- Create a GitHub Action that runs yamllint
- Fix yamllint warnings
  - Double quote the `on` keyword
`on` -> `"on"`
  - Fix comment indentation

Tested locally by using `act`. Added random things to random YAMLs in order to make the lint fail and it indeed failed.

### CODE REVIEW

File: .tekton/tasks/yaml-lint.yaml [hunk 1]
```diff
@@ -1,50 +0,0 @@
...
-          if yq '.spec?.steps[] | .script' $task | grep -q '\$(params\.'; then
-            FAILED_TASKS="$FAILED_TASKS $task"
-          fi
-        done
-        if [ -n "$FAILED_TASKS" ]; then
-          echo "Tasks contains params in script section (https://tekton.dev/docs/pipelines/tasks/#substituting-in-script-blocks)"
-          echo $FAILED_TASKS | tr ' ' '\n' | sort
-          exit 1
-        fi
-      workingDir: $(workspaces.shared-workspace.path)/source
```
Comment (chmeliik) [inline .tekton/tasks/yaml-lint.yaml:50]: The yamllint task also did something entirely not yamllint-y (^ this check)

We need to preserve this check

File: .github/workflows/yaml-lint.yaml [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+"on":
+  pull_request:
+    branches: [main]
+jobs:
+  yamllint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Lint YAML files
+        run: yamllint .
```
Comment (chmeliik) [inline .github/workflows/yaml-lint.yaml:12]: > create a GitHub Action that installs yamllint and then runs it

It looks like it doesn't even install it. Is yamllint just pre-installed on GH runners?

Comment (tnevrlka) [reply .github/workflows/yaml-lint.yaml:12]: Yes, it is

Comment (chmeliik) [reply .github/workflows/yaml-lint.yaml:12]: Nice. Could you just update the commit message?

**Review by chmeliik — APPROVED** [review]

**Pull request merged**